### PR TITLE
Add lanes()

### LINKED ISSUE #151: Missing function for number of lanes
One of my projects is currently using `packed_simd`, and uses a compile-time feature flag to choose a vector size. I tried porting this to `stdsimd`, which worked perfectly, apart from a missing function, to get the current number of lanes for a given simd type. `packed_simd` has the const fn [`lanes`](https://rust-lang.github.io/packed_simd/packed_simd_2/struct.Simd.html#method.lanes) which I used in this context. It would be great to have this also exposed for all types in this crate, and it should be rather trivial to implement.

Edit: It seems the trait item T::LANES is available, but this is awkward to find  in the documentation, and requires the `Vector` trait to be in scope. I am not sure whether adding the `lanes()` function is worth it for the ease of documentation.

### PR DESCRIPTION
**Pull request by mulimoen**
Adds the lanes function for easily getting the number of lanes of a SIMD element

Fixes #151

### CODE REVIEW

**Comment by calebzulawski** [comment]
I wonder if it makes more sense to do this or have a const associated with the struct.  This makes it easier to pull from an instance of a vector, of course.

**Comment by Lokathor** [comment]
could have both, but associated consts are "weirder" to use, so if there's just one then this should be it.

**Comment by mulimoen** [comment]
I'd say it would be best to include both to enable the most flexibility for users

File: crates/core_simd/src/vector.rs [hunk 1]
```diff
@@ -24,6 +24,13 @@ where
     LaneCount<LANES>: SupportedLaneCount,
     T: SimdElement,
 {
+    const LANES: usize = LANES;
```
Comment (calebzulawski) [inline crates/core_simd/src/vector.rs:27]: Missing `pub` and documentation

Comment (mulimoen) [reply crates/core_simd/src/vector.rs]: Added `pub` and a comment

**Comment by workingjubilee** [comment]
So I don't want to bikeshed excessively, but should this be `len`, in sync with e.g. Vec, slice, and other containers?

**Comment by calebzulawski** [comment]
Yeah, it may make more sense calling it `len`.  It's worth noting that arrays don't have a `len` function at all, you need to use `as_slice` or similar.

**Comment by Lokathor** [comment]
well if you call array.len you'll fail the method lookup, automatically deref to a slice, and then the slice will have it. So *effectively* arrays have this.

But i think lanes makes sense because generally one should not be "looking inside" a simd vector horizontally that much

**Comment by mulimoen** [comment]
I don't feel strongly for `len` over `lanes`, but `lanes` might be more intuitive as the documentation is heavy on the use of `LANES`.

**Comment by workingjubilee** [comment]
Fair enough. If we change it we should change it elsewhere, then.

**Pull request merged**